### PR TITLE
Move content of proj_experimental.h to proj.h

### PR DESCRIPTION
**Pull request by rouault**
Those functions have been used for ages by GDAL 3.X. They are not so much "experimental" nowadays.

No backward compatibility issue as proj_experimental.h includes proj.h

### CODE REVIEW

**Comment by mwtoews** [comment]
Should any of these be expanded into docs/source/development/reference/{datatypes,functions}.rst?

**Comment by rouault** [comment]
> Should any of these be expanded into docs/source/development/reference/{datatypes,functions}.rst?

goot point. Adressed by additional commit

File: docs/source/development/reference/functions.rst [hunk 1]
```diff
@@ -1047,3 +1047,6 @@ will return an error when used with functions of this section.
    :project: doxygen_api
    :content-only:
 
+.. doxygengroup:: advanced_cpp_binding
```
Comment (mwtoews) [inline docs/source/development/reference/functions.rst]:
```suggestion
.. doxygengroup:: advanced_cpp_functions
```

and then `datatypes.rst` would have similar but reference `advanced_cpp_types`

File: src/proj.h [hunk 2]
```diff
@@ -1504,6 +1504,688 @@ double PROJ_DLL proj_coordinate_metadata_get_epoch(PJ_CONTEXT *ctx,
...
 /**@}*/
 
+/* ------------------------------------------------------------------------- */
+/* Binding in C of advanced methods from the C++ API                         */
+/*                                                                           */
+/* Manual construction of CRS objects.                                       */
+/* ------------------------------------------------------------------------- */
+
+/**
+ * \defgroup advanced_cpp_binding Binding in C of advanced methods from the C++
```
Comment (mwtoews) [inline src/proj.h]:
```suggestion
 * \defgroup advanced_cpp_types Binding in C of advanced methods from the C++
```
and perhaps a slightly different title too.

Then start another group `advanced_cpp_functions` around L1592

**Review by mwtoews — COMMENTED** [review]
Suggest splitting `advanced_cpp_binding` into `advanced_cpp_types` and `advanced_cpp_functions` so they can be put into each of the `datatypes.rst` and `functions.rst` docs respectively. This name pair mirror the existing two groups: `iso90111_types` and `iso90111_functions`.

**Comment by rouault** [comment]
> Suggest splitting `advanced_cpp_binding` into `advanced_cpp_types` and `advanced_cpp_functions`

done, by amending last commit

**Comment by mwtoews** [comment]
Thanks, rendered docs looks good! A few comments:

- Looking at the datatype docs, it is curious to see [`struct PJ_AXIS_DESCRIPTION`](https://osgeo-proj--4019.org.readthedocs.build/en/4019/development/reference/datatypes.html#c.PJ_AXIS_DESCRIPTION) and `struct PJ_PARAM_DESCRIPTION` do not show public members similar to [`struct PROJ_CELESTIAL_BODY_INFO`](https://osgeo-proj--4019.org.readthedocs.build/en/4019/development/reference/datatypes.html#c.PROJ_CELESTIAL_BODY_INFO). Is this because they needs to start with `/** \brief ...`?
- Potentially `@since 9.4` could be added everywhere. But I wouldn't, since the docs are now versioned.
- Should there be section headers in the rst docs? Currently the former and advanced content are all under the "C API for ISO-19111 functionality" sections. The doxygengroup titles don't seem to show up. This is less relevant for the handful of new datatypes, but there are many new advanced functions that are added. Maybe insert a "Advanced C API" section for functions.rst?

**Comment by snowman2** [comment]
> Potentially `@since` 9.4 could be added everywhere. But I wouldn't, since the docs are now versioned.

It is nice to have that so you don't have to dig through different versions of the docs to find out when it was added. But, not critical.

**Comment by rouault** [comment]
> * Looking at the datatype docs, it is curious to see [`struct PJ_AXIS_DESCRIPTION`](https://osgeo-proj--4019.org.readthedocs.build/en/4019/development/reference/datatypes.html#c.PJ_AXIS_DESCRIPTION) and `struct PJ_PARAM_DESCRIPTION` do not show public members similar to [`struct PROJ_CELESTIAL_BODY_INFO`](https://osgeo-proj--4019.org.readthedocs.build/en/4019/development/reference/datatypes.html#c.PROJ_CELESTIAL_BODY_INFO). Is this because they needs to start with `/** \brief ...`?

I've tried different things, but don't manage to fix. Must be some Doxygen/Breathe/Sphinx bug

> Potentially @since 9.4 could be added everywhere. But I wouldn't, since the docs are now versioned.

I've added a global remark "Available in :file:`proj.h` since PROJ 9.4. Previously were available in :file:`proj_experimental.h`."

> Should there be section headers in the rst docs? 

Done, but it is too deep in the arborescence to show up in the left menu

**Comment by mwtoews** [comment]
Looks good, almost there. I just noticed a mix-up in the new docs from #3559 with these suggested corrections:
```diff
diff --git a/src/iso19111/c_api.cpp b/src/iso19111/c_api.cpp
index 17df96ee2..a353fe179 100644
--- a/src/iso19111/c_api.cpp
+++ b/src/iso19111/c_api.cpp
@@ -5208,7 +5208,7 @@ PJ *proj_create_conversion_two_point_equidistant(
  * linear_unit_conv_factor).
  * Angular parameters are expressed in (ang_unit_name, ang_unit_conv_factor).
  *
- * @deprecated. Replaced by proj_create_conversion_tunisia_mining_grid
+ * @since 9.2
  */
 PJ *proj_create_conversion_tunisia_mining_grid(
     PJ_CONTEXT *ctx, double center_lat, double center_long,
@@ -5242,7 +5242,7 @@ PJ *proj_create_conversion_tunisia_mining_grid(
  * linear_unit_conv_factor).
  * Angular parameters are expressed in (ang_unit_name, ang_unit_conv_factor).
  *
- * @since 9.2
+ * @deprecated Replaced by proj_create_conversion_tunisia_mining_grid
  */
 PJ *proj_create_conversion_tunisia_mapping_grid(
     PJ_CONTEXT *ctx, double center_lat, double center_long,
```
with the move from experimental header, should `proj_create_conversion_tunisia_mapping_grid` be dropped?

**Comment by rouault** [comment]
> I just noticed a mix-up in the new docs from #3559 with these suggested corrections:

done

> with the move from experimental header, should proj_create_conversion_tunisia_mapping_grid be dropped?

I would perhaps keep that for a PROJ 10

File: src/iso19111/c_api.cpp [hunk 3]
```diff
@@ -5242,7 +5242,7 @@ PJ *proj_create_conversion_tunisia_mining_grid(
  * linear_unit_conv_factor).
  * Angular parameters are expressed in (ang_unit_name, ang_unit_conv_factor).
  *
- * @since 9.2
+ * @deprecated. Replaced by proj_create_conversion_tunisia_mining_grid
```
Comment (mwtoews) [inline src/iso19111/c_api.cpp]:
```suggestion
 * @deprecated Replaced by proj_create_conversion_tunisia_mining_grid
```

Comment (rouault) [reply src/iso19111/c_api.cpp]: done. thanks

**Review by mwtoews — APPROVED** [review]

**Pull request merged**